api/handlers/dataexplorerhandler.py

```diff
@@ -251,6 +251,91 @@ def _parse_request(self, request_type='search'):
 
         return return_type, modified_filters, search_string
 
+    @require_login
+    def get_type_ahead_values(self):
+        """
+        Return list of type ahead values for a key given a value 
+        that the user has already started to type in for the value of 
+        a custom string field
+        """
+        user_value = self.request.json_body['value']
+        custom_field = self.request.json_body['field_name']
+        modified_filters = []
+        # Add permissions filter to list if user is not requesting all data
+        if not self.request.json_body.get('all_data', False):
+            modified_filters.append({'term': {'permissions._id': self.uid}})
```

Comment:
I think we would want this filter applied all the time - a user might put something they'd rather no one else see in a custom field. Retrieving all results does't allow a user to see more than a few descriptive fields of results.